java/ruleXtract/src/main/java/uk/ac/cam/eng/extraction/hadoop/features/lexical/TTableServer.java

```diff
@@ -269,73 +264,18 @@ public void close() throws IOException {
-		@Parameter(names = { "--ttable_s2t_server_port" }, description = "TTable source-to-target server port")
-		public String ttable_s2t_server_port = "4949";
-
-		@Parameter(names = { "--ttable_s2t_host" }, description = "TTable source-to-target host name")
-		public String ttable_s2t_host = "localhost";
-
-		@Parameter(names = { "--ttable_t2s_server_port" }, description = "TTable target-to-source server port")
-		public String ttable_t2s_server_port = "9494";
-
-		@Parameter(names = { "--ttable_t2s_host" }, description = "TTable target-to-source host name")
-		public String ttable_t2s_host = "localhost";
-
-		@Parameter(names = { "--ttable_server_template" }, description = "TTable target-to-source host name", required = true)
-		public String ttable_server_template;
-
-		@Parameter(names = { "--ttable_direction" }, description = "TTable direction for the lexical model ('s2t' or 't2s')", required = true)
-		public String ttable_direction;
-
-		@Parameter(names = { "--ttable_language_pair" }, description = "TTable language pair for the lexical model (e.g. 'en2ru' or 'ru2en')", required = true)
-		public String ttable_language_pair;
-
-		@Parameter(names = { "--provenance" }, description = "Comma-separated list of provenances", required = true)
-		public String provenance;
-
-		@Parameter(names = { "--min_lex_prob" }, description = "Minimum probability for a Model 1 entry. Entries with lower probability are discarded.")
-		public String min_lex_prob = "0";
-	}
-
-	public int run(String[] args) throws IllegalArgumentException,
+	public static void main(String[] args) throws IllegalArgumentException,
 			IllegalAccessException, IOException, InterruptedException {
-		TTableServerParameters params = new TTableServerParameters();
-		JCommander cmd = new JCommander(params);
-
-		try {
-			cmd.parse(args);
-			Configuration conf = getConf();
-			Util.ApplyConf(cmd, "", conf);
-			boolean source2Target;
-			if (params.ttable_direction.equals("s2t")) {
-				source2Target = true;
-			} else if (params.ttable_direction.equals("t2s")) {
-				source2Target = false;
-			} else {
-				throw new RuntimeException("Unknown direction: " + args[2]);
-			}
-			try (TTableServer server = new TTableServer()) {
-				server.setup(conf, params.ttable_language_pair, source2Target);
-				server.startServer();
-				System.err.println("TTable server ready on port: "
-						+ server.serverSocket.getLocalPort());
-				Thread.sleep(24 * 60 * 60 * 1000); // Sleep for 24 hours
-			}
-		} catch (ParameterException e) {
-			System.err.println(e.getMessage());
-			cmd.usage();
+		CLI.TTableServerParameters params = new CLI.TTableServerParameters();
+		Util.parseCommandLine(args, params);
+
+		try (TTableServer server = new TTableServer()) {
+			server.setup(params);
+			server.startServer();
+			System.err.println("TTable server ready on port: "
+					+ server.serverSocket.getLocalPort());
+			Thread.sleep(24 * 60 * 60 * 1000); // Sleep for 24 hours
```

Comment:
Does this mean that the server will be 24 hours available tops and then it will shut down? Is this regardless of the server activity? Could you please document here the behaviour?